src/api/common/MessageDispatcher.ts

```diff
@@ -38,29 +38,29 @@ export class WorkerTransport<RequestType, ResponseType> implements Transport<Req
 		return this._worker.postMessage(message)
 	}
 
-	setMessageHandler(handler: (arg0: Message<ResponseType>) => unknown) {
+	setMessageHandler(handler: (message: Message<ResponseType>) => unknown) {
 		this._worker.onmessage = (ev: MessageEvent) => handler(downcast(ev.data))
 	}
 }
 
 export class Request<T> {
-	type: "request"
+	readonly type: "request"
 	readonly requestType: T
-	id: string
-	args: any[]
+	readonly id: string
+	readonly args: any[]
 
 	constructor(type: T, args: ReadonlyArray<unknown>, requestId: string | null = null) {
 		this.type = "request"
 		this.requestType = type
 		this.id = requestId ?? _createRequestId()
-		this.args = Array.from(args)
+		this.args = args.slice()
```

Comment:
this seems like reintroducing old code? What's the rationale for not using Array.from anymore?

Reply:
`args` is already an array, there's no need to treat it as an iterable of unknown type